### PR TITLE
[iOS] Add dismiss keyboard button to EditorHandler

### PR DESCRIPTION
**Pull request by jsuarezruiz**
### Description of Change ###

Add pending events and dismiss keyboard button (on iPhone) to EditorHandler.

### PR Checklist ###

- [x] Targets the correct branch 
- [ ] Tests are passing (or failures are unrelated)

#### Does this PR touch anything that might affect accessibility?
- No

### CODE REVIEW

**Review by PureWeen — CHANGES_REQUESTED** [review]
AutoSize is already implemented at the controls level

https://github.com/dotnet/maui/pull/1202

**Comment by jsuarezruiz** [comment]
> AutoSize is already implemented at the controls level
> 
> #1202

Right, but this include:

`public event EventHandler? FrameChanged;`

To detect when the native control Frame changes and update the scroll.

File: src/Core/src/Handlers/Editor/EditorHandler.iOS.cs [hunk 1]
```diff
@@ -11,7 +11,29 @@ public partial class EditorHandler : ViewHandler<IEditor, MauiTextView>
 	{
 		static readonly int BaseHeight = 30;
 
-		protected override MauiTextView CreatePlatformView() => new MauiTextView();
+		protected override MauiTextView CreatePlatformView()
+		{
+		var nativeEditor =	new MauiTextView();
+
+			if (DeviceInfo.Idiom == DeviceIdiom.Phone)
```
Comment (PureWeen) [inline src/Core/src/Handlers/Editor/EditorHandler.iOS.cs:18]: Can this PR just be the code here? and then we can discuss the `AutoSize` changes here?

https://github.com/dotnet/maui/pull/4850

**Review by PureWeen — CHANGES_REQUESTED** [review]

File: src/Core/src/Handlers/Editor/EditorHandler.iOS.cs (hunk 1 above)
Comment (jsuarezruiz) [reply src/Core/src/Handlers/Editor/EditorHandler.iOS.cs:18]: Yes, updated the PR to only include the changes in the PlatformView. @PureWeen Could you review it again, please?

File: src/Core/src/Handlers/Editor/EditorHandler.iOS.cs [hunk 2]
```diff
@@ -1,15 +1,29 @@
...
 namespace Microsoft.Maui.Handlers
 {
 	public partial class EditorHandler : ViewHandler<IEditor, MauiTextView>
 	{
-		protected override MauiTextView CreatePlatformView() => new MauiTextView();
+		protected override MauiTextView CreatePlatformView()
+		{
+			var platformEditor = new MauiTextView();
+
+			if (DeviceInfo.Idiom == DeviceIdiom.Phone)
```
Comment (PureWeen) [inline src/Core/src/Handlers/Editor/EditorHandler.iOS.cs:16]: @jsuarezruiz do we need to check the Idiom everywhere in the code that we use this `InputAccessoryView` ?

**Review by PureWeen — CHANGES_REQUESTED** [review]

File: src/Core/src/Handlers/Editor/EditorHandler.iOS.cs (hunk 2 above)
Comment (jsuarezruiz) [reply src/Core/src/Handlers/Editor/EditorHandler.iOS.cs:16]: We want this behavior on iOS, not Desktop but we don't need the same check everywhere (TimePicker etc).

**Review by PureWeen — APPROVED** [review]

**Pull request merged**